tests/integration/workflows/helpers/workflow_utils.py

```diff
@@ -22,22 +22,31 @@
 
 from __future__ import absolute_import, division, print_function
 
+import uuid
 from copy import deepcopy
 
-from invenio_workflows import workflow_object_class
+from invenio_workflows import workflow_object_class, Workflow
 
 
-def build_workflow(workflow_data, data_type='hep', extra_data=None, **kwargs):
-    workflow_object = workflow_object_class.create(
-        data=workflow_data,
-        data_type=data_type,
-        extra_data={
+def build_workflow(workflow_data, data_type='hep', extra_data=None, status=None, **kwargs):
+    extra_data = extra_data or {}
+    if 'source_data' not in extra_data:
+        extra_data = {
             'source_data': {
                 'data': deepcopy(workflow_data),
-                'extra_data': extra_data or {},
+                'extra_data': extra_data,
             }
-        },
+        }
+    wf = Workflow(name='article', extra_data=extra_data, uuid=uuid.uuid4())
```

Comment:
why do you need this? can't you pass the `uuid` and `name` to `workflow_object_class.create`?

Reply:
Unfortunately No. `workflow_object_class` is actually [invenio_workflows_files.api.WorkflowObject](https://github.com/inveniosoftware-contrib/invenio-workflows-files/blob/1e2b07ea4e46f89e3ca30211919e8cf7680c030c/invenio_workflows_files/api.py#L53) Which is mostly actually [invenio_workflows.api.WorkflowObject](https://github.com/inveniosoftware-contrib/invenio-workflows/blob/892a9c3cc2134566bcd575c43e55865d6381a4a0/invenio_workflows/api.py#L47) and in `create` we are not even able to properly create `Workflow` object which is holding data about workflow type, workflow class etc. Which is required now to properly filter-out workflows. Proper way of creating WF with `Workflow` Object properly assigned to is it only through [`save`](https://github.com/inveniosoftware-contrib/invenio-workflows/blob/892a9c3cc2134566bcd575c43e55865d6381a4a0/invenio_workflows/api.py#L98-L136) method which is doing this if if_workflow of already created `Workflow` object is passed. It's messy. It took me yesterday some time to understand how this works in the backend :/

Reply:
I can explain it on zoom if you would like to ;)

Reply:
It's ok, I can wait for the presentation :wink: